src/selectors/url-state.js

```diff
@@ -109,9 +107,12 @@ export const getMarkersSearchString: Selector<string> = (state) =>
   getProfileSpecificState(state).markersSearchString;
 export const getNetworkSearchString: Selector<string> = (state) =>
   getProfileSpecificState(state).networkSearchString;
-
 export const getSelectedTab: Selector<TabSlug> = (state) =>
   getUrlState(state).selectedTab;
+export const getInvertCallstack: Selector<boolean> = (state) =>
+  getSelectedTab(state) === 'calltree' &&
+  getProfileSpecificState(state).invertCallstack;
+
```

Comment:
Is this what you mentioned during the standup? That's a bit slower now but will be faster later?

With this, I believe that we'll recompute the call node info when changing the tab, is that right?

See [this deploy preview using a profile with a lot of samples](https://deploy-preview-4804--perf-html.netlify.app/public/280e10708c0660695ef5af145c53179d29d9796c/calltree/?globalTrackOrder=0w4&invertCallstack&thread=4&timelineType=category&v=10) / [production](https://profiler.firefox.com/public/280e10708c0660695ef5af145c53179d29d9796c/calltree/?globalTrackOrder=0w4&invertCallstack&thread=4&timelineType=category&v=10)

I can see that it's slow to move from Flamegraph to Call tree, but not the other way around, therefore my guess is that the action of inverting itself is slow, not the computation of the call node table.

Therefore what do you think of computing the inverted call tree in a separate selector (so that it's memoized), and return one or the other in `getFilteredThread`? `getFilteredThread` itself wouldn't use `createSelector` so that we wouldn't need to compute the inverted tree if we don't need it:
```js
export const getFilteredThread: Selector<Thread> = state => {
  const invertCallStack = UrlState.getInvertCallStack(state);
  return UrlState.getInvertCallStack(state)
    ? _getInvertedThread(state)
    : _getImplementationAndSearchFilteredThread(state);
}
```

Reply:
Oh I was thinking of other slowdowns when I said this in the standup; slowdowns which I'll introduce in other upcoming patches. (Specifically: computing an "inverted ordering" of non-inverted call nodes)

Thank you for this suggestion, this is actually quite helpful! I had similar problems where I wanted to do something conditionally but didn't want to eagerly compute both inputs, and didn't have an idea for how to do that with createSelector... but you're totally right, there's no reason I'd have to use createSelector here.

I'll incorporate this suggestion.

Reply:
> therefore my guess is that the action of inverting itself is slow, not the computation of the call node table.

By the way, the computation of the inverted call node table is slower than the computation of the non-inverted call node table. That's because the inverted stackTable has a lot more nodes (O(n^2)) and the inverted call node table also has a lot of nodes.
So both the computation of the inverted thread and the computation of the inverted call node table are slow.

But this change will cache both; getCallNodeInfo already has a 2-element cache, so it won't need to recompute if its input thread changes between inverted and non-inverted if it has computed both versions before.